### PR TITLE
feat: ✨ Tweet related widgets

### PR DESCRIPTION
**Pull request by Roaa94**
Adds small components used in a parent Tweet component like metrics, username, display name, ..etc.

### CODE REVIEW

**Comment by jenshor** [comment]
It looks like this build did not deploy on Widgetbook Cloud. Can you add an action that pushes feature branches on pull-requests?

File: packages/core/lib/core.dart [hunk 1]
```diff
@@ -2,5 +2,4 @@
 library core;
 
 export './src/styles/styles.dart';
-export './src/tweet/tweet.dart';
```
Comment (jenshor) [inline packages/core/lib/core.dart:5]: Looks odd to me that this was removed. Is this intentional?

File: packages/core/lib/src/widgets/tweet/detailed_tweet_metrics.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+/// Likes, Retweets, and Quote Tweet metrics for a detailed tweet
+class DetailedTweetMetrics extends StatelessWidget {
+  /// Creates a new instance of [DetailedTweetMetrics]
+  const DetailedTweetMetrics({
+    super.key,
+    this.retweetsCount = 0,
+    this.quoteTweetsCount = 0,
+    this.likesCount = 0,
+  });
+
```
Comment (jenshor) [inline packages/core/lib/src/widgets/tweet/detailed_tweet_metrics.dart:13]: For this Widget, the alignment of numbers and text seems to be a bit off 
image

File: packages/core/lib/src/widgets/tweet/retweets.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      activeIcon: const Icon(Icons.repeat),
+      iconSize: widget.iconSize,
+      activeColor: AppColors.success,
+      onPressed: () {
+        // Todo: This should open a dropdown of "Retweet/Quote Tweet"
+        // Only Retweet activates the metric
+        // Quote Tweet should open a dialog to submit a tweet
+        setState(() {
+          _isActive = !_isActive;
+        });
```
Comment (jenshor) [inline packages/core/lib/src/widgets/tweet/retweets.dart:51]: We should create Issues if you have open Todo's merged into staging. GitHub has a syntax to parse these todos, which makes them easily accessible.

File: packages/core/lib/src/widgets/user/username.dart [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+class Username extends StatelessWidget {
+  /// Creates a new instance of [Username]
+  const Username({
+    super.key,
+    required this.value,
+    this.color,
+  });
+
+  /// Username value
+  final String value;
```
Comment (jenshor) [inline packages/core/lib/src/widgets/user/username.dart:16]: If the username is `widgetbook_io` the widget currently shows `widgetbook_io` without the `@` sign. Therefore, I think the comment is a bit confusing.

File: packages/core/pubspec.yaml [hunk 5]
```diff
@@ -9,6 +9,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
+  intl: ^0.17.0
```
Comment (jenshor) [inline packages/core/pubspec.yaml:12]: `intl` package should not be part of this PR, but to keep the overhead of managing this low, we can keep it in.

**Review by jenshor — COMMENTED** [review]
Looks great 🎉🚀

I just had a few questions 😊

File: packages/core/lib/core.dart (hunk 1 above)
Comment (Roaa94) [reply packages/core/lib/core.dart:5]: Yes. It's now inside `src/widgets/widgets.dart` 
I have some concerns in general for the folder structure that I would love to discuss with you.

File: packages/core/lib/src/widgets/tweet/detailed_tweet_metrics.dart (hunk 2 above)
Comment (Roaa94) [reply packages/core/lib/src/widgets/tweet/detailed_tweet_metrics.dart:13]: The line height in this font seems very inconsistent. I set it to 1 for all styles and I think this fixed this widget. However, maybe we can consider a different font if the problem persists?

File: packages/core/lib/src/widgets/tweet/retweets.dart (hunk 3 above)
Comment (Roaa94) [reply packages/core/lib/src/widgets/tweet/retweets.dart:51]: Should we use this or do you have something else in mind?
https://github.com/marketplace/actions/todo-to-issue

File: packages/core/lib/src/widgets/user/username.dart (hunk 4 above)
Comment (Roaa94) [reply packages/core/lib/src/widgets/user/username.dart:16]: Agreed. Fixed this.

File: packages/core/pubspec.yaml (hunk 5 above)
Comment (Roaa94) [reply packages/core/pubspec.yaml:12]: I'm using it in the `src/widgets/tweet/metric_text.dart` widget to format the numbers with `NumberFormat` (e.g. `10000` => `10k`)

Should we not depend on it?

**Review by jenshor — APPROVED** [review]
Line hight seems still to be a bit off, but it already looks a lot better. Let's get this merged now and I'll create an issue so we can revisit it later.

File: packages/core/pubspec.yaml (hunk 5 above)
Comment (jenshor) [reply packages/core/pubspec.yaml:12]: Oh didn't realize that and confused it with localization 😀
